cmd/soroban-rpc/internal/db/sqlmigrations/03_events.sql

```diff
@@ -5,7 +5,7 @@ CREATE TABLE events(
     id                TEXT PRIMARY KEY,
     ledger_sequence   INTEGER NOT NULL,
     application_order INTEGER NOT NULL,
-    contract_id       BLOB,
+    contract_id       TEXT,
```

Comment:
Why is this? The DB shouldn't be storing text ids (i.e. hex) it should be binary, so that it occupies less space.

Reply:
In fact, I think this could be a `BLOB(32)`

Reply:
Now I am noticing that the id field is also text.  Why is that? (sorry I didn't notice before)

Reply:
id field here is basically a cursor id (TOID) of event. That is why it is stored as TEXT

Reply:
[TOID is a 64-bit number](https://github.com/stellar/go/blob/master/toid/main.go#L17-L18); can we have a 64-bit integer column in sqlite?

Reply:
Sorry my bad, I was storing the string representation of it. INTEGER makes sense.

Reply:
@psheth9 sorry for the late reply! that comment is basically saying if you have a collection of ledgers, operations, and transactions, the toid cannot uniquely identify an item in that collection because the ledger `n` has the same toid as the first operation in ledger `n` and the first transaction in ledger `n`. In that scenario, you don't know if it refers to the ledger, the transaction, or the operation.

However, the reason that comment does not apply in your situation is that you have a table consisting of only 1 type: events. If all objects have the same type in a collection then the toid will be unique.